temporal-sdk/src/main/java/io/temporal/internal/sync/POJOActivityTaskHandler.java

```diff
@@ -328,8 +332,13 @@ public DynamicActivityImplementation(DynamicActivity activity) {
       }
       inboundCallsInterceptor.init(context);
       try {
-        EncodedValues args = new EncodedValues(input, dataConverter);
-        Object result = inboundCallsInterceptor.execute(new Object[] {args});
+        EncodedValues encodedValues = new EncodedValues(input, dataConverter);
+        Object[] args = new Object[] {encodedValues};
+
+        Object result =
+            inboundCallsInterceptor.execute(
+                new ActivityInboundCallsInterceptor.ActivityInput(
+                    new Header(info.getHeader()), args));
```

Comment:
This approach would only solve the problem of getting headers, but what about setting them?

Reply:
What do you mean by "setting them"? The interceptor has full control over header content before passing it to the next interceptor.

Reply:
We already pass headers to Activity execution inside ```ExecuteActivityParameters#ScheduleActivityTaskCommandAttributes.Builder#setHeader```.
For that to actually do any useful work Interceptors or ContextPropagators have to actually add the headers first.